### PR TITLE
some documentation and style fixes to smooth_l1_loss

### PR DESCRIPTION
**Pull request by bdhirsh**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #45588 remove beta defaulting in smooth_l1_loss_backward. added to the bc whitelist
* **#45587 some documentation and style fixes to smooth_l1_loss**

Differential Revision: [D24024313](https://our.internmc.facebook.com/intern/diff/D24024313)

### CODE REVIEW

File: aten/src/ATen/native/Loss.cpp [hunk 1]
```diff
@@ -296,17 +296,21 @@ Tensor soft_margin_loss(
 }
 
 Tensor smooth_l1_loss(const Tensor& input, const Tensor& target, const int64_t reduction, double beta) {
-  if (beta <= 0)
+  TORCH_CHECK(beta >= 0, "smooth_l1_loss does not support negative values for beta.")
```
Comment (mrshenli) [inline aten/src/ATen/native/Loss.cpp:299]: Do we have tests to cover these errors?

Comment (bdhirsh) [reply aten/src/ATen/native/Loss.cpp:299]: just added one

File: test/test_nn.py [hunk 2]
```diff
@@ -7032,6 +7032,10 @@ def test_l1_loss_correct(self):
                 torch.nn.L1Loss()(input, torch.zeros_like(input)),
                 input.abs().mean())
 
+    def test_smoothl1loss_negative_beta_not_supported(self):
+        with self.assertRaises(RuntimeError):
```
Comment (mrshenli) [inline test/test_nn.py:7036]: nit: you can use `assertRaisesRegex` to also verify that the error message is as expected.

**Review by mrshenli — APPROVED** [review]
LGTM!

File: torch/nn/modules/loss.py [hunk 3]
```diff
@@ -780,6 +780,9 @@ class SmoothL1Loss(_Loss):
 
     beta is an optional parameter that defaults to 1.
 
+    Note: When beta is set to 0, this is equivalent to we call out directly to :class:`L1Loss`.
```
Comment (gchanan) [inline torch/nn/modules/loss.py:783]: this doesn't look grammatically correct, did you mix up two messages?  I would expect something like:

"Note: When beta is set to 0, this is equivalent to calling L1Loss."

Also, is that the correct note doc format?

Comment (bdhirsh) [reply torch/nn/modules/loss.py:783]: thanks, fixing :/ 

I skimmed a few other files containing docs to find the `:class:` anotation- but is there a command that I can run locally to test the formatting?
